engine/access/rpc/rate_limit_test.go

```diff
@@ -117,38 +116,27 @@ func (suite *RateLimitTestSuite) SetupTest() {
 	block := unittest.BlockHeaderFixture()
 	suite.snapshot.On("Head").Return(block, nil)
 
-	finalizationDistributor := pubsub.NewFinalizationDistributor()
-
-	var err error
-	finalizedHeaderCache, err := synceng.NewFinalizedHeaderCache(suite.log, suite.state, finalizationDistributor)
-	require.NoError(suite.T(), err)
-
 	rpcEngBuilder, err := NewBuilder(suite.log, suite.state, config, suite.collClient, nil, suite.blocks, suite.headers, suite.collections, suite.transactions, nil,
 		nil, suite.chainID, suite.metrics, suite.metrics, 0, 0, false, false, apiRateLimt, apiBurstLimt, suite.me)
 	assert.NoError(suite.T(), err)
-	suite.rpcEng, err = rpcEngBuilder.WithLegacy().WithFinalizedHeaderCache(finalizedHeaderCache).Build()
+	suite.rpcEng, err = rpcEngBuilder.WithLegacy().Build()
 	assert.NoError(suite.T(), err)
-	unittest.AssertClosesBefore(suite.T(), suite.rpcEng.Ready(), 2*time.Second)
-
+	suite.ctx, suite.cancel = irrecoverable.NewMockSignalerContextWithCancel(suite.T(), context.Background())
+	suite.rpcEng.Start(suite.ctx)
 	// wait for the server to startup
-	assert.Eventually(suite.T(), func() bool {
-		return suite.rpcEng.UnsecureGRPCAddress() != nil
-	}, 5*time.Second, 10*time.Millisecond)
+	unittest.AssertClosesBefore(suite.T(), suite.rpcEng.Ready(), 2*time.Second)
 
 	// create the access api client
 	suite.client, suite.closer, err = accessAPIClient(suite.rpcEng.UnsecureGRPCAddress().String())
 	assert.NoError(suite.T(), err)
 }
 
 func (suite *RateLimitTestSuite) TearDownTest() {
+	suite.cancel()
```

Comment:
fwiw, these nil checks should probably be kept, and all of the `assert.NoError` in `SetupTest` be converted to `require.NoError`

Reply:
It wasn't obvious to me why the nil checks were needed. Is this the reasoning to keep them? 👇 
- `SetupTest` might fail before these fields are set
- `TearDownTest` will still be run, and cause a nil pointer exception
- That will halt the whole package of tests, instead of just the one case where `SetupTest` failed

Reply:
Yea, exactly. The nil pointer exceptions hide the original error so it's harder to debug

Reply:
Makes sense - I've added applied the suggestion here: https://github.com/onflow/flow-go/pull/4313/commits/2be6f0d88e7365291a93608394e7a2d1aec85401